src/Rx/Weno/NarcoticRxController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Narcotic Prescription Controller
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @copyright Copyright (c) 2019 Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Rx\Weno;
+
+require_once "../../interface/globals.php";
+
+use OpenEMR\Core\Controller;
```

Comment:
What is the above Controller doing for this code?
(note we were thinking of removing that class since it didn't do anything, so want to make sure you don't need it)

Reply:
It does nothing at the moment. I only added because it was there and if in the future it held something useful, it would already be there.  I will remove it.

Reply:
Hi @bradymiller this class is very important for future development, I hope you can reconsider the removing idea about this class.

Reply:
hi @zerai ,
That was removed in a bunch of unused code here:
https://github.com/openemr/openemr/pull/2869
(it resulted in a bunch of dead end code that was not used)(feel free to bring back the class if use it for something; also would be a good opportunity to ensure it is following best practices for symfony 4.4 since the best practices seem to quickly evolve for symfony)

Reply:
I know about changes in #2869.  
That PR/merge made me crazy :-), but at that time and even today I don't have a big picture about the internal mechanism of OpenEMR (initialization process, HTTP Request/Response) In my feelings we have lost something in #2869, but as a developer, I should respond or comment a PR with techs arguments and not with feelings.
In any case, we can reintroduce the 'service.yml' file in a feature PR when we really need it.

Reply:
About this class, [we have spent some words](https://community.open-emr.org/t/questions-from-an-openemr-newbie/13326/3) about the 2 service container argument and the absent use  of dependency injection.

An abstract/base Controller doesn't solve this problem in a permanent way but can mitigate/centralize object construction, reduce indiscriminate access to global vars, improve general testability and other stuff.

Reply:
No reason for community to spend time updating/maintaining code that does nothing. Again, though, door is always open to bring it back in if it actually does something.